### PR TITLE
Basic sparseml integration for openpifpaf

### PR DESCRIPTION
**Pull request by corey-nm**
Testing plan: Training command runs, and verified sparsity levels change in output from loggers

### CODE REVIEW

File: src/sparseml/pytorch/openpifpaf/utils.py [hunk 1]
```diff
@@ -12,8 +12,116 @@
...
+                "checkpoint_recipe": str(self.manager),
+                "epoch": epoch,
+                "meta": self.model_meta_data,
+            },
+            filename,
+        )
+        LOG.info("model written: %s", filename)
+
+        if final:
+            sha256_hash = hashlib.sha256()
```
Comment (dbogunowicz) [inline src/sparseml/pytorch/openpifpaf/utils.py:120]: Is this something we are doing now by default? Generating the model string from sha256?

**Review by dbogunowicz — APPROVED** [review]
Looking good to me. The correctness of sparsification can be tested with the integration tests as we discussed today.

File: src/sparseml/pytorch/openpifpaf/utils.py (hunk 1 above)
Comment (corey-nm) [reply src/sparseml/pytorch/openpifpaf/utils.py:120]: SHould probably make a not of this - I just copied this function from their trainer class and updated the stuff in the checkpoint dictionary. So this is their logic. We can probably change it to something else if we don't want to keep that

**Review by bfineran — COMMENTED** [review]
looks clean

File: src/sparseml/pytorch/openpifpaf/utils.py [hunk 2]
```diff
@@ -12,8 +12,116 @@
...
+            self.checkpoint_shell.load_state_dict(model.state_dict())
+            model_to_save = self.checkpoint_shell
+
+        filename = "{}.epoch{:03d}".format(self.out, epoch)
+        LOG.debug("about to write model")
+        torch.save(
+            {
+                "model": model_to_save,
+                "state_dict": model_to_save.state_dict(),
+                "checkpoint_recipe": str(self.manager),
```
[the remark this answers is not in the document]
Comment (corey-nm) [reply src/sparseml/pytorch/openpifpaf/utils.py:111]: yep

File: src/sparseml/pytorch/openpifpaf/README.md [hunk 3]
```diff
@@ -2,7 +2,8 @@
```
[the remark this answers is not in the document]
Comment (bfineran) [reply src/sparseml/pytorch/openpifpaf/README.md:2]: Yes, we should let the integration dependencies handle it

**Review by bfineran — APPROVED** [review]

**Pull request merged**